code/processes/tailer.q

```diff
@@ -46,9 +47,40 @@ getprocs:{[x;y]
         }
 
 .servers.register[.servers.procstab;.tailer.tailreadertypes;1b]
+.servers.register[.servers.procstab;.tailer.eodtypes;1b]
 
 \d .
 
+/- eod - send end of day message to main EOD process
+endofday:{[pt;processdata]
+    .lg.o[`eod;"end of day message received - ",spt:string pt];
+
+    /- trigger final save down from tailer
+    endofdaysave[processdata];
+
+    /- find handle to send message to eod process
+    eodp:exec w from .servers.getservers[`proctype;.tailer.eodtypes;()!();1b;0b];
+
+    /- exit early if no eod process connected
+    if[0=count eodp;.lg.e[`connection;"no connection to the ",(string .tailer.eodtypes)," could be established, failed to send end of day message"];:()];
+
+    /- send procname to eod process so it loads correct TDB
+    procname:.proc.procname;
+    neg[first eodp](`endofday;pt;procname);
+    .lg.o[`eod;"end of day message sent to eod process"];
+    };
+
+endofdaysave:{[processdata]
+  /- function to flush remaining in-memory data to disk when end of day message is received
+  .lg.o[`save;"saving the ",(", " sv string tl:.wdb.tablelist[],())," table(s) to disk"];
+  currp:first exec distinct end from .ds.access where end<>0N;
+  .wdb.datastripeendofperiod[currp;.z.p;processdata[],(enlist `p)!enlist .z.p];
+  .lg.o[`savefinish;"finished saving remaining data to disk"];
+  };
+
+/- add endofday to tailer namespace and overwrite .wdb.endofday function
+.tailer.endofday:.wdb.endofday:endofday;
```

Comment:
Why does wdb.endofday need to be updated here? Is this a hangover from having wdb.q loaded into tailer.q?

Reply:
Yep, because the endofday functions from wdb.q are loaded in I wanted to make sure that the only endofday function present in the tailer is the one defined here